x/supplier/keeper/msg_server_update_param.go

```diff
@@ -3,35 +3,47 @@ package keeper
 import (
 	"context"
 
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
 	suppliertypes "github.com/pokt-network/poktroll/x/supplier/types"
 )
 
 // UpdateParam updates a single parameter in the proof module and returns
 // all active parameters.
 func (k msgServer) UpdateParam(ctx context.Context, msg *suppliertypes.MsgUpdateParam) (*suppliertypes.MsgUpdateParamResponse, error) {
 	if err := msg.ValidateBasic(); err != nil {
-		return nil, err
+		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
 	if k.GetAuthority() != msg.Authority {
-		return nil, suppliertypes.ErrSupplierInvalidSigner.Wrapf("invalid authority; expected %s, got %s", k.GetAuthority(), msg.Authority)
+		return nil, status.Error(
+			codes.InvalidArgument,
+			suppliertypes.ErrSupplierInvalidSigner.Wrapf(
+				"invalid authority; expected %s, got %s",
+				k.GetAuthority(), msg.Authority,
+			).Error(),
+		)
 	}
 
-	// TODO_UPNEXT(@bryanchriswhite, #612): uncomment & add a min_stake case.
-
-	//params := k.GetParams(ctx)
+	params := k.GetParams(ctx)
 
 	switch msg.Name {
+	case suppliertypes.ParamMinStake:
+		params.MinStake = msg.GetAsCoin()
 	default:
-		return nil, suppliertypes.ErrSupplierParamInvalid.Wrapf("unsupported param %q", msg.Name)
+		return nil, status.Error(
+			codes.InvalidArgument,
+			suppliertypes.ErrSupplierParamInvalid.Wrapf("unsupported param %q", msg.Name).Error(),
+		)
+	}
```

Comment:
```suggestion
	}
	
	if err := params.Validate(); err != nil {
		return nil, status.Error(codes.InvalidArgument, err.Error())	
	}
```

_TODO: Also update in "adding_params.md" doc._